### PR TITLE
fix: OrgUnit validation in event query [DHIS2-11755]

### PR DESCRIPTION
**Pull request by abyot**

### CODE REVIEW

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/events/EventExportTests.java [hunk 1]
```diff
@@ -146,7 +146,7 @@ public void beforeEach()
     Stream<Arguments> shouldUseCorrectScopeWhenOuIsProvided()
     {
         return Stream.of(
-            Arguments.of( "OU: root", "SELECTED", rootOu, false, null ),
+            Arguments.of( "OU: root", "SELECTED", rootOu, true, Arrays.asList( rootOu ) ),
```
Comment (abyot) [inline dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/events/EventExportTests.java:149]: @vilkg see if this is a correct combination.

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/events/EventExportTests.java [hunk 2]
```diff
@@ -238,13 +238,16 @@ public void shouldUseCorrectScopeWhenOuModeIsProvided( String description, Strin
     private Stream<Arguments> shouldReturnSingleEvent()
     {
         return Stream.of( new Arguments[] {
-            Arguments.of( "PROGRAM: event, OU: search, shouldReturn: false", events.get( searchOu ), false ),
+            Arguments.of( "PROGRAM: event, OU: search, shouldReturn: false", events.get( searchOu ), true ),
```
Comment (abyot) [inline dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/events/EventExportTests.java:241]: @vilkg, see if this is a correct combination.

**Review by stian-sandvold — APPROVED** [review]

File: dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/events/EventExportTests.java (hunk 1 above)
Comment (abyot) [reply dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/events/EventExportTests.java]: this is fixed now.

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java [hunk 3]
```diff
@@ -807,4 +803,20 @@ private boolean isNull( ProgramStage programStage )
         return programStage == null || programStage.getProgram() == null;
     }
 
+    private boolean isOrgUnitAccessible( User user, Program program, OrganisationUnit orgUnit )
+    {
+        // always allow if these are null. Internal process?
```
Comment (teleivo) [inline dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java:808]: What do you mean by `Internal process?`

This logic looks like it is the same for DefaultTrackerAccessManager and AbstractEventService? Code duplicated in 2 places is not per se a bad thing if they do not mean exactly the same thing. If they are likely to evolve differently. If the logic in DefaultTrackerAccessManager and AbstractEventService isOrgUnitAccessible should stay in sync then it is dangerous to have it in two places. They could get out of sync if one forgets to update both. Which one is it? 1. Looks the same but has different meaning 2. looks and means the same thing and should stay in sync 😄

**Review by teleivo — COMMENTED** [review]
Very readable change for what felt like a complicated issue to me @abyot 😄 
I added one question. Also now that I see how small the change looks like I wonder if it would be possible to unit test these changes? Have a few unit tests that cover the different branches and only a few e2e ones. To speed up running tests 😄

**Review by Philip-Larsen-Donnelly — APPROVED** [review]

**Review by vilkg — APPROVED** [review]
lgtm :)

File: dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java (hunk 3 above)
Comment (abyot) [reply dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/trackedentity/DefaultTrackerAccessManager.java]: yes, the accessible check should be centralized.

**Pull request merged**